### PR TITLE
feat: Support log level when syncing IDE logpoints in VS Code Plugin

### PR DESCRIPTION
**Pull request by jasonborg**

### CODE REVIEW

File: snapshot_dbg_extension/src/logLevelPicker.ts [hunk 1]
```diff
@@ -17,6 +21,27 @@ export async function pickLogLevel(): Promise<string | undefined> {
...
+
     return selection?.label.split(' ')[0].trim();
 }
+
+export async function pickLogLevelNewlyCreated(): Promise<string | undefined> {
+    return pickLogLevel('Select Log Level');
+}
+
+export async function pickLogLevelSyncedFromIDE(path: string, line: number): Promise<string | undefined> {
+    return pickLogLevel(`Select Log Level For Logpoint Being Synced From the IDE: ${path}:${line}`);
```
Comment (mctavish) [inline snapshot_dbg_extension/src/logLevelPicker.ts:46]: This is pretty verbose.  We can probably remove "Being Synced From the IDE".

File: snapshot_dbg_extension/src/adapter.ts [hunk 2]
```diff
@@ -64,6 +64,7 @@ export class SnapshotDebuggerSession extends DebugSession {
     private setVariableType: boolean = false;
     private userPreferences: UserPreferences;
     private isDeferredInitializationDone = false;
+    private activeInitializePathCount = 0;
```
Comment (mctavish) [inline snapshot_dbg_extension/src/adapter.ts:67]: It's worth adding a comment in here; the intention behind this variable is not obvious to me.

File: snapshot_dbg_extension/src/adapter.ts [hunk 3]
```diff
@@ -365,12 +372,17 @@ export class SnapshotDebuggerSession extends DebugSession {
         debugLog(response.body);
         this.sendResponse(response);
 
-        if (flushServerBreakpointsToIDE) {
-            flushServerBreakpointsToIDE();
+        if (finalizeInitializePath) {
```
Comment (mctavish) [inline snapshot_dbg_extension/src/adapter.ts:375]: Come to think of it, it's also worth tossing a comment here to indicate why we have this finalize step.

**Review by mctavish — APPROVED** [review]
Looks good, could use a couple small comments to hint to future readers why this specific setup was used.

File: snapshot_dbg_extension/src/logLevelPicker.ts (hunk 1 above)
Comment (jasonborg) [reply snapshot_dbg_extension/src/logLevelPicker.ts]: Good call.

**Pull request merged**